packages/typespec-ts/test/azureModularIntegration/routes.spec.ts

```diff
@@ -0,0 +1,232 @@
+        try {
+            // Should encode the path parameter because skipUrlEncoding is explicitly set to false
+            await client.pathParameters.reservedExpansion.template("foo/bar baz", {
+                requestOptions: {
+                    skipUrlEncoding: false
+                }
+            });
+            assert.fail("Should fail this case if path parameter is not encoded");
+        }
+        catch (error: any) {
+            assert.equal(error.statusCode, 404);
+            assert.isTrue(error?.request?.url.includes("/foo%2Fbar%20ba"));
+        }
+    });
+    it("Routes_PathParameters_ReservedExpansion_annotation", async () => {
+        const result = await client.pathParameters.reservedExpansion.annotation("foo/bar baz");
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_SimpleExpansion_Standard_primitive", async () => {
+        const result = await client.pathParameters.simpleExpansion.standard.primitive("a");
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_SimpleExpansion_Standard_array", async () => {
+        const result = await client.pathParameters.simpleExpansion.standard.array(["a", "b"]);
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_SimpleExpansion_Standard_record", async () => {
+        const result = await client.pathParameters.simpleExpansion.standard.record({ a: 1, b: 2 });
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_PathExpansion_Standard_primitive", async () => {
+        const result = await client.pathParameters.pathExpansion.standard.primitive("a");
+        assert.isUndefined(result);
+    });
+
+    it("Routes_PathParameters_PathExpansion_Standard_array", async () => {
+        const result = await client.pathParameters.pathExpansion.standard.array(["a", "b"]);
+        assert.isUndefined(result);
+    });
+
+    it("Routes_PathParameters_PathExpansion_Standard_record", async () => {
+        const result = await client.pathParameters.pathExpansion.standard.record({ a: 1, b: 2 });
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_LabelExpansion_Standard_primitive", async () => {
+        const result = await client.pathParameters.labelExpansion.standard.primitive("a");
+        assert.isUndefined(result);
+    });
+
+    it("Routes_PathParameters_LabelExpansion_Standard_array", async () => {
+        const result = await client.pathParameters.labelExpansion.standard.array(["a", "b"]);
+        assert.isUndefined(result);
+    });
+    it("Routes_PathParameters_LabelExpansion_Standard_record", async () => {
+        const result = await client.pathParameters.labelExpansion.standard.record({ a: 1, b: 2 });
+        assert.isUndefined(result);
+    });
+    // TODO: enable these cases when issue is fixed https://github.com/microsoft/typespec/issues/5430
+    describe.skip("Skip these cases", () => {
```

Comment:
pls note some cases are pending on spector: https://github.com/microsoft/typespec/issues/5430

Reply:
@v-jiaodi Could you help verify if the version is upgraded and these cases are fixed?